packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/model/inheritance/single-discriminator/src/Generated/Models/Eagle.cs

```diff
@@ -0,0 +1,27 @@
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace _Type.Model.Inheritance.SingleDiscriminator.Models
+{
+    public partial class Eagle : Bird
+    {
+        public Eagle(int wingspan) : base("eagle", wingspan) => throw null;
+
+        internal Eagle(IList<Bird> friends, IDictionary<string, Bird> hate, Bird partner, int wingspan, IDictionary<string, BinaryData> serializedAdditionalRawData) : base("eagle", wingspan, serializedAdditionalRawData) => throw null;
```

Comment:
In derived class, the full construtor/deserialzieContructor will contain discriminator parameter Kind in autorest.csharp(see following), but now in MGC, we remove the discriminator property parameter.
```
internal Eagle(string kind, int wingspan, IDictionary<string, BinaryData> serializedAdditionalRawData, IList<Bird> friends, IDictionary<string, Bird> hate, Bird partner) : base(kind, wingspan, serializedAdditionalRawData)
```
Is this change expected? and we know that in codegen, we will identify a full constructor by the number of the parameters, if we remove kind, it may cause some problem.